### PR TITLE
Updates for Qt5

### PR DESCRIPTION
**Pull request by zquestz**
This PR fixes all essential deprecation warnings and a bunch of small warning nits.

Code compiles and works well using Qt5.

The only unresolved warnings are unused variables and params. I can live with those.

### CODE REVIEW

File: src/ctransaction.cpp [hunk 1]
```diff
@@ -123,6 +123,7 @@ void CTransaction::addData(const char * data) {
         case 110:
         case 115:
             newParameter.type = TYPE_STRING;
+            break;
```
Comment (zquestz) [inline src/ctransaction.cpp:126]: Can you confirm this break is correct? It looked like it to me.

File: src/main.cpp [hunk 2]
```diff
@@ -9,16 +9,16 @@
...
 
 qint32 main(qint32 argc, char *argv[])
 {
-    FILE * debugLog = fopen("log.txt", "a");
-    if(debugLog)
-    {
-        fwrite("\n", 1, 1, debugLog);
-        fclose(debugLog);
-    }
+    // FILE * debugLog = fopen("log.txt", "a");
```
Comment (zquestz) [inline src/main.cpp:16]: I removed this debug file, as it was never used and was just empty.

File: Obsession.pro [hunk 3]
```diff
@@ -112,8 +112,7 @@ FORMS += src/mainwindow.ui \
     src/dialogbroadcast.ui \
     src/dialogpostlinearnews.ui
 INCLUDEPATH += src/
-RESOURCES += resources/resources.qrc \
-    resources/icons1.qrc \
+RESOURCES += resources/icons1.qrc \
```
Comment (zquestz) [inline Obsession.pro:115]: resources.qrc was listed twice, removed the duplicate.

File: src/cdownload.cpp [hunk 4]
```diff
@@ -212,7 +212,7 @@ void CDownload::startDownloading() {
     if(connection->pSocket.isValid() && connection->pSocket.state() == QAbstractSocket::ConnectedState) {
         socket.connectToHost(connection->pSocket.peerAddress(), connection->pSocket.peerPort()+1, QIODevice::ReadWrite);
         socket.waitForConnected();
-        if(socket.state() == !QAbstractSocket::ConnectedState) {
+        if(socket.state() != QAbstractSocket::ConnectedState) {
```
Comment (zquestz) [inline src/cdownload.cpp:215]: This was to fix an enum to bool warning, pretty sure this was the intention.

File: src/cdownload.cpp [hunk 5]
```diff
@@ -296,14 +296,14 @@ void CDownload::serverReady() {
 
     qint32 nameSize = forkSize - 72;
 
-    bool hxd = false;
```
Comment (zquestz) [inline src/cdownload.cpp:299]: We never use this, so I just commented it out for now.

File: src/ctransaction.cpp [hunk 6]
```diff
@@ -224,7 +224,9 @@ char * CTransaction::bytes() {
         qint16 plen = qToBigEndian(pParameters[i]->length());
         memcpy(b+24+offset, &plen, 2);
 
-        qint16 sval, ival;
+        qint16 sval;
+        qint32 ival;
```
Comment (zquestz) [inline src/ctransaction.cpp:228]: Wanted to double check that this change was valid. It was trying to read 4 bytes from a var that only stored two... so I updated the type, which I believe is correct.

File: src/ctransaction.cpp (hunk 1 above)
Comment (tjohnman) [reply src/ctransaction.cpp:126]: This should be correct, as it is for the Chat Subject parameter.

**Comment by tjohnman** [comment]
Thank you for the PR. But this would break Qt 4 compatibility, which we specifically use to maintain builds for older macs. QStandardPaths for example did not exist until Qt 5.

**Comment by zquestz** [comment]
Did you want to maintain a Qt5 branch?

**Comment by zquestz** [comment]
Also, can you look at the last two comments, want to make sure those are right. =)

File: src/ctransaction.cpp (hunk 6 above)
Comment (tjohnman) [reply src/ctransaction.cpp:228]: Yes, this was a bug for sure.

File: src/cdownload.cpp (hunk 4 above)
Comment (tjohnman) [reply src/cdownload.cpp:215]: Absolutely.

File: src/ctransaction.cpp (hunk 6 above)
Comment (zquestz) [reply src/ctransaction.cpp:228]: Awesome, we should try to get the bug fixes into the main branch. Then the leftovers I can keep in a QT5 branch for now.

**Comment by tjohnman** [comment]
> Did you want to maintain a Qt5 branch?

We actually downgraded from Qt 5 to 4 to make it compatible on older machines. As long as Qt 4 works on modern ones I see no point really.

**Comment by zquestz** [comment]
Gotcha, then I will just maintain my patch for the AUR!

**Comment by tjohnman** [comment]
Thank you. I will implement some of the tweaks and fixes you did. Closing this now.

**Pull request closed without merging**